pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+
+			// Check that the namespace does not exist already
+			ensure!(!<Namespaces<T>>::contains_key(&namespace), Error::<T>::NamespaceAlreadyExists);
+
+			// burn NOVA from account's balance. The amount is set in Config.
+			let amount: <T as pallet_balances::Config>::Balance =
+				<T as Config>::NamespacePrice::get().saturated_into();
+			<pallet_balances::Pallet<T> as fungible::Mutate<T::AccountId>>::burn_from(
+				&who, amount,
+			)?;
+
+			<Namespaces<T>>::insert(&namespace, &who);
+
+			Self::deposit_event(Event::NamespaceCreated { who, namespace });
+
+			Ok(())
+		}
+
+		/// Delete a new **Namespace**.
+		///
+		/// This also deletes all the aliases linked to this namespace
+		///
+		/// - `namespace`: namespace to delete
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn delete_namespace(
+			origin: OriginFor<T>,
+			namespace: BoundedVec<u8, <T as pallet::Config>::NameLimit>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(!namespace.len().is_zero(), Error::<T>::InvalidInput);
+
+			let namespace = namespace.into_inner();
+
+			// Check that the namespace exists
+			ensure!(<Namespaces<T>>::contains_key(&namespace), Error::<T>::NamespaceAlreadyExists);
+			// check that the caller is the owner of the namespace
+			let owner = <Namespaces<T>>::get(&namespace).ok_or(Error::<T>::NamespaceNotFound)?;
+			ensure!(&who == &owner, Error::<T>::NotAllowed);
+
+			<Namespaces<T>>::remove(&namespace);
+			let _ = <Aliases<T>>::clear_prefix(&namespace, u32::MAX, None);
+
+			Self::deposit_event(Event::NamespaceDeleted { namespace });
+
+			Ok(())
+		}
+
+		/// Transfer the **Namespace ownership** to another AccountId.
+		///
+		/// Only the owner of the Namespace can execute this.
+		///
+		/// - `namespace`: namespace to create
+		/// - `new_owner`: the AccountId to transfer ownership to
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn transfer_namespace(
+			origin: OriginFor<T>,
+			namespace: BoundedVec<u8, <T as pallet::Config>::NameLimit>,
+			new_owner: T::AccountId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(who != new_owner, Error::<T>::NotAllowed);
+			ensure!(!namespace.len().is_zero(), Error::<T>::InvalidInput);
+
+			let namespace = namespace.into_inner();
+
+			// Check that the caller is the owner of the namespace
+			let owner = <Namespaces<T>>::get(&namespace).ok_or(Error::<T>::NamespaceNotFound)?;
+			ensure!(&who == &owner, Error::<T>::NotAllowed);
+
+			<Namespaces<T>>::insert(&namespace, &new_owner);
+
+			Self::deposit_event(Event::NamespaceTransferred {
+				namespace,
+				from: who,
+				to: new_owner,
+			});
+
+			Ok(())
+		}
+
+		/// Create an **Alias** that links to a target asset.
+		///
+		/// Only the owner of the asset can create its alias.
+		///
+		/// - `namespace`: namespace to create
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn create_alias(
```

Comment:
add `create_root_alias`? or we scrap that feature?

Reply:
I made a `update_alias_target`, callable only from root, which is what you called `update_root_alias` in the issue #226 . 

You mean to implement a `create_root_alias` function that would create an alias without checking for the ownership of the related `LinkTarget` (since root has not an AccountId, so if we want to check ownership of the asset I would need to pass a T::AssetId as a parameter)?

Reply:
Creating an alias without related namespace would mean creating an entry in the `Aliases StorageDoubleMap` without key1 (key1 = namespace, key2 = alias). 
So, the options here are to either insert root aliases in that StorageDoubleMap with a "fake" namespace, which would indicate that they are "root" aliases, or create another storage dedicated to root aliases.
What do you prefer?

Also, two considerations/questions:
1. this would also mean that root aliases without namespace will be unique
2. what would be the scope of this root aliases? Can they be assigned to a namespace in the future?

Reply:
Yeah I think we can establish a proprietary namespace. Say `frag`. And add to to config. That should fix all issues. 👍

Reply:
Done.